src/components/accessories/billFilterForm/BillFilterForm.tsx

```diff
@@ -0,0 +1,167 @@
+      onSubmit(formattedValues);
+    },
+  });
+
+  const { setFieldValue, handleBlur } = formik;
+
+  const dateFieldHandleOnChange = useCallback(
+    (fieldName: string) => (value: any) => {
+      setFieldValue(fieldName, value);
+      formik.handleSubmit();
+    },
+    [setFieldValue, formik.handleSubmit]
+  );
+
+  const isValid = (fieldName: string): boolean => {
+    return has(formik.touched, fieldName) && has(formik.errors, fieldName);
+  };
+
+  const getErrorText = (fieldName: string): string => {
+    return has(formik.touched, fieldName)
+      ? (get(formik.errors, fieldName) as string)
+      : "";
+  };
+
+  const onBlurCallback = useCallback(
+    (fieldName: string) =>
+      (e: React.FocusEvent<HTMLDivElement>, value: string) => {
+        formik.handleBlur(e);
+        setFieldValue(fieldName, value);
+        formik.handleSubmit();
+      },
+    [setFieldValue, handleBlur]
+  );
+
+  return (
+    <>
+      <div className={"filterBillForm " + className}>
+        <form className="filterBillForm__form" onSubmit={formik.handleSubmit}>
+          <div className="row start-sm center-xs">
+            <div className="fullWidth filterBillForm__item">
+              <DateField
+                theme={"light"}
+                fieldName="fromDate"
+                fieldValue={formik.values.fromDate}
+                disableFuture={false}
+                format="dd/MM/yyyy"
+                isValid={isValid("fromDate")}
+                errorText={getErrorText("fromDate")}
+                label={t("bill.fromdate")}
+                onChange={dateFieldHandleOnChange("fromDate")}
+              />
+            </div>
+            <div className="fullWidth filterBillForm__item">
+              <DateField
+                fieldName="toDate"
+                fieldValue={formik.values.toDate}
+                disableFuture={false}
+                theme="light"
+                format="dd/MM/yyyy"
+                isValid={isValid("toDate")}
+                errorText={getErrorText("toDate")}
+                label={t("bill.todate")}
+                onChange={dateFieldHandleOnChange("toDate")}
+              />
+            </div>
+          </div>
+          <div className="row start-sm center-xs">
+            <div className="fullWidth filterBillForm__item">
+              <AutocompleteField
+                theme={"light"}
+                fieldName="patient"
+                fieldValue={formik.values.patient}
+                label={t("bill.patient")}
+                isValid={isValid("patient")}
+                errorText={getErrorText("patient")}
+                onBlur={onBlurCallback("patient")}
+                isLoading={searchStatus === "LOADING"}
+                options={renderOptions(patientSearchResults)}
+                onInputChange={handlePatientSearchChange}
+                freeSolo={true}
+              />
+            </div>
+          </div>
+        </form>
```

Comment:
I think that we should add a button in order to submit the form

Reply:
It think it is right. I add it